packages/php-table-backend-utils/src/Table/TableMultiKindInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\TableBackendUtils\Table;
+
+interface TableMultiKindInterface
```

Comment:
Představoval sem si, že to bude SnowflakeTableKind jak to měls a nebude to v interface, ale proč, ne uvidíme jak to dopadne v ostatních backendoch pak :) 
Nevím jestli sa mě teda líbý MultiKind to je dost divné :D 
- TableWithKindInterface 
- WithTableKindInterface
- SupportsTableKindInterface
:thinking: Co na to name master @tomasfejfar ?

Reply:
Nebo sa vyprdnůt na interface a udělat to jako součást toho Table definition interface. Ostatní implementace možů mět 
```
getKind(){return TableKind::TABLE}
```
nic sa tím nepokazí a bude to celé jednodušší.

Reply:
- naming is hard, muzu zmenit. Ten `SupportsTableKindInterface` se mi asi libi. 
- chtel jsem to mit jako interface abych v connection pak mohl udelat `instance of XXXInterface` misto checkovani N backendu (ano, ted jen snflk)
- nemame nad tim abstract class, jen interface, takze jsem nechtel lozit do vsech ostatnich `XXXTableDefinition` trid a pridavat to tam. to bych radeji udelal tu abstraktni tridu

Reply:
jo a puvodne jsem to mel jako `SnowflakeTableKind`, ale pak jsi mi vnuknul myslenky ze v connection potrebujeme nejaky unifikovany kind, tak jsem ho vykopl tady

Reply:
No právě když ten interface ostraníš tak ten check nemusíš dělat místo toho, že máš všude v connection instaceod TableKindInterface tam budeš mět proště $defitnion->getTableKind() a je to, já bych to teda odstranil a žádné abstraktní třídy nedělal to je cesta do pekel.

Reply:
Ok ok, jdu to tak udelat.

Reply:
Nevím, jestli bych to popisoval jako "table kind", když je to view :) není to spíš backendObjectType?

Reply:
Sme trochu narazili na limit toho rozhraní tím, že TableDefinition má kind a kromě table to može byt view a external table :D Ale jirka tady kopíruje víceméně SNFLK api.
ikdyž na druhů stranu information_schema má TABLE_TYPE to može byt BASE_TABLE,VIEW,... takže bych možná reflektoval tady to, páč je to SQL standard :) Takže `getTableType(): TableType`? :)